purchases/src/main/kotlin/com/revenuecat/purchases/common/HTTPClient.kt

```diff
@@ -36,11 +37,13 @@ import java.net.URLConnection
 import java.util.Date
 import kotlin.time.Duration
 
+@Suppress("LongParameterList")
 internal class HTTPClient(
     private val appConfig: AppConfig,
     private val eTagManager: ETagManager,
     private val diagnosticsTrackerIfEnabled: DiagnosticsTracker?,
     val signingManager: SigningManager,
+    private val deviceCache: DeviceCache,
```

Comment:
Instead of injecting this huge type, how about injecting a `UserContextProvider` interface?

Otherwise ripping out DeviceCache from here in the future might get harder.

Reply:
Hmm what else would you add to the `UserContextProvider`? If it would be only for getting the storefront, I would probably have something more specific right now, like `StorefrontProvider` or something like that? Wdyt?

Reply:
Changed in https://github.com/RevenueCat/purchases-android/pull/1484/commits/2efb8b52dfdc744ebb41ab481ef1c23423bed78a